src/pybind/mgr/telemetry/module.py

```diff
@@ -293,9 +293,51 @@ def get_active_channels(self):
 
     def gather_device_report(self):
         try:
-            return self.remote('devicehealth', 'gather_device_report')
+            time_format = self.remote('devicehealth', 'get_time_format')
         except:
             return None
+        cutoff = datetime.utcnow() - timedelta(hours=self.interval * 2)
+        min_sample = cutoff.strftime(time_format)
+
+        devices = self.get('devices')['devices']
+
+        res = {}
+        for d in devices:
+            devid = d['devid']
+            try:
+                m = self.remote('devicehealth', 'get_recent_device_metrics',
+                                devid, min_sample)
+            except:
+                continue
+
+            # anonymize host id
+            try:
+                host = d['location'][0]['host']
+            except:
+                continue
+            anon_host = self.get_store('host-id/%s' % host)
+            if not anon_host:
+                anon_host = str(uuid.uuid1())
+                self.set_store('host-id/%s' % host, anon_host)
+            m['host_id'] = anon_host
+
+            # anonymize device id
+            (vendor, model, serial) = devid.split('_')
+            anon_devid = self.get_store('devid-id/%s' % devid)
+            if not anon_devid:
+                anon_devid = '%s_%s_%s' % (vendor, model, uuid.uuid1())
+                self.set_store('devid-id/%s' % devid, anon_devid)
+
+            self.log.info('devid %s / %s, host %s / %s' % (devid, anon_devid,
+                                                           host, anon_host))
+
+            # anonymize the smartctl report itself
+            for k in ['serial_number']:
```

Comment:
the list seems unnecessary (were there more purge items once?) and m.pop('serial_number') will do it without the test (returns the value, but you can ignore the return).  Slightly more Pythonic.